ManageIQ/manageiq
# Add Ironic Controls for OpenStack Infrastructure

This patch adds menu options allowing admins to introspect hosts,
provide hosts, and set hosts to manageable state.

## Steps for Testing/QA 

Requires OSP 10 or an upstream TripleO deployment with Mistral and TripleO workflows.

Introspection
1. Register a new node. Or select a node that is in manageable state and it's introspected attribute is set to false.
2. Go to host/node and select "Introspect Node" under "Configuration".
3. A message indicating introspect has been queued will be displayed. If a node is not in manageable state, an error message will be displayed.
4. After introspection has completed, a refresh is initiated.
5. After refresh is complete, you should see that the introspected attribute is now true.

Provide
1. Select a node that is in manageable state.
2. Go to host/node and select "Provide Node" under "Configuration".
3. A message indicating provide has been queued will be displayed. If the node is not in manageable state, an error message will be displayed.
4. After provide has completed, a refresh is initiated.
5. After refresh is complete, you should see that the provision_state has changed to available. Note: 

Set Node to Manageable
1. Select a node that is in available state.
2. Go to host/node and select "Set Node to Manageable" under "Configuration".
3. A message indicating "Set Node to Manageable" has been queued will be displayed. If the node is not in available state, an error message will be displayed.
4. After "Set Node to Manageable" has completed, a refresh is initiated.
5. After refresh is complete, you should see that the provision_state has changed to manageable.
